### ISSUE TITLE
Thief crashing on startup after 5 seconds of black screen

### ISSUE THREAD

**Issue by LucasSnatiago** [HansKristian-Work/vkd3d-proton#2224]
I'm trying to run the game [Thief](https://www.epicgames.com/store/product/thief-5bb95f) inside heroic games launcher. After starting the game the launcher runs just fine, but when I hit the play button, the game starts a black screen and crashes after 5 seconds.

### Software information
[Thief](https://www.epicgames.com/store/product/thief-5bb95f)
Using the default ultra settings

### System information
- GPU: RX 7600 XT
- Driver: Mesa 24.2.7-1 
- Wine version: 1732404742 experimental-bleeding-edge-9.0-137858-20241123-p0b1b64-w321a2d-d9e1bc1-vff1286
- VKD3D-Proton version: Bleeding Edge commit -> 4f549155c5625acacd6b2f04d84d6cc41cbbaa11

### Log files
[steam-heroic-0.log](https://github.com/user-attachments/files/17895740/steam-heroic-0.log)
[steam-heroic-ThiefvWwV2.log](https://github.com/user-attachments/files/17895741/steam-heroic-ThiefvWwV2.log) -> In the end of this file is written:
```
20004.695:0128:fixme:d3dcompiler:D3DCompile2 Ignoring flags 0x800.
20004.695:0128:err:d3dcompiler:D3DCompile2 Failed to compile shader, vkd3d result -4.
20004.695:0128:err:d3dcompiler:D3DCompile2 Shader log:
20004.695:0128:err:d3dcompiler:D3DCompile2     ..\..\Engine\\Shaders\Compute\DoFBlurBGCS.hlsl:35:51: E5000: Invalid subscript "Length".
20004.695:0128:err:d3dcompiler:D3DCompile2     ..\..\Engine\\Shaders\Compute\DoFBlurBGCS.hlsl:35:5: E5002: Can't implicitly convert from <error type> to int.
20004.695:0128:err:d3dcompiler:D3DCompile2     ..\..\Engine\\Shaders\Compute\DoFBlurBGCS.hlsl:42:83: E5000: Invalid subscript "Length".
20004.695:0128:err:d3dcompiler:D3DCompile2     ..\..\Engine\\Shaders\Compute\DoFBlurBGCS.hlsl:43:83: E5000: Invalid subscript "Length".
20004.695:0128:err:d3dcompiler:D3DCompile2     ..\..\Engine\\Shaders\Compute\DoFBlurBGCS.hlsl:44:83: E5000: Invalid subscript "Length".
20004.695:0128:err:d3dcompiler:D3DCompile2     ..\..\Engine\\Shaders\Compute\DoFBlurBGCS.hlsl:45:83: E5000: Invalid subscript "Length".
20004.695:0128:err:d3dcompiler:D3DCompile2     ..\..\Engine\\Shaders\Compute\DoFBlurBGCS.hlsl:46:1: E5005: Function "GroupMemoryBarrierWithGroupSync" is not defined.
20004.695:0128:err:d3dcompiler:D3DCompile2 

```

**Comment by mbriar** [comment]
Install `d3dcompiler_47` with winetricks to the prefix.

**Comment by LucasSnatiago** [comment]
Interesting, I tried installing ```d3dcompiler_47```, but the error persisted. Then I installed all d3dcompilers available: ```d3dcompiler_47, d3dcompiler_42, d3dcompiler_43, d3dcompiler_46```. With that the game now runs, but with some visual glitches. Can it be correlated to the fact I installed more than one version of d3dcompiler?

**Comment by mbriar** [comment]
I don't think so, it will only use the d3dcompiler version it needs. What glitches are we talking about, can you post a screenshot?

**Comment by LucasSnatiago** [comment]
Better than a screenshot. I recorded a video: [Thief Visual Glitches - Blinking objects](https://youtu.be/xpmr4iVFF6o).

**Comment by mbriar** [comment]
If you're talking about the flickering stuff, that's just the game having shitty zfighting. You can find a bunch of videos on youtube with it happening on windows as well, e.g.: https://www.youtube.com/watch?v=zkuJMsUNeu8

Side note, but this game is using D3D11 and not 12, so it couldn't be a vkd3d-proton issue either way.

**Comment by LucasSnatiago** [comment]
Thank you for all the help!
